# Screen Time Data: Productivity and Attention Span

El dataset “Screen Time Data: Productivity and Attention Span” (https://www.kaggle.com/datasets/muhammadalirazazaidi/screen-time-data-productivity-and-attention-span?utm_source=chatgpt.com) reúne respuestas de encuesta de 200 participantes con el objetivo de analizar cómo el tiempo frente a pantallas se relaciona con la productividad y la capacidad de atención. Además de medir el uso diario de dispositivos digitales, el conjunto de datos incorpora variables demográficas y patrones de comportamiento que permiten estudiar posibles asociaciones entre hábitos digitales y desempeño cognitivo.

Columnas:
*   Age Group: grupo etario del participante.
*   Gender: género del participante.
*   Education Level: nivel educativo alcanzado.
*   Occupation: ocupación o actividad principal.
*   Average Screen Time: promedio diario de tiempo frente a pantallas.
*   Device: dispositivo preferido para el uso digital.
*   Screen Activity: actividad principal realizada en pantalla.
*   App Category: categoría de aplicaciones más utilizadas.
*   Screen Time Period: momento del día en que se concentra el uso de pantalla.
*   Environment: entorno de trabajo preferido.
*   Productivity: nivel de productividad autopercibido al realizar tareas basadas en pantalla.
*   Attention Span: duración de la atención autopercibida durante tareas digitales.
*   Work Strategy: estrategia de trabajo preferida (por ejemplo, multitarea o trabajo enfocado).
*   Notification Handling: forma en que el participante gestiona notificaciones.
*   Usage of Productivity Apps: indica si utiliza herramientas digitales para mejorar su productividad.

Este conjunto de datos permite explorar patrones de consumo digital y analizar cómo distintos perfiles demográficos, hábitos de uso y estrategias de trabajo podrían estar asociados con variaciones en la atención y la productividad. En las siguientes secciones se desarrollará un análisis exploratorio para identificar tendencias, relaciones y posibles implicancias del uso intensivo de dispositivos digitales en el rendimiento cognitivo.

In [ ]:
import pandas as pd
import numpy as np
import plotly.express as px

In [ ]:
# Se carga el dataset
df = pd.read_csv("data.csv")

# Se realiza una vista rápida para verificar los datos
df.head()

,Unnamed: 0,Age Group,Gender,Education Level,Occupation,Average Screen Time,Device,Screen Activity,App Category,Screen Time Period,Environment,Productivity,Attention Span,Work Strategy,Notification Handling,Usage of Productivity Apps
0,0,18–24,Male,Undergraduate,Student,More than 10,Smartphone,"Entertainment (gaming, streaming, social media...","Social Media (e.g., Facebook, Instagram, Linke...",Evening (6 PM–10 PM),Quite workplace,Moderately productive,10–30 minutes,Take regular breaks,Check them briefly and resume my work,"Yes, but i did not find them of any help"
1,1,18–24,Male,Undergraduate,Professional,8-10,Smartphone,"Entertainment (gaming, streaming, social media...","Streaming (e.g., YouTube, Netflix)",Late night (10 PM–6 AM),Quite workplace,Moderately productive,More than 1 hour,"None, i prefer to work without any strategies",Ignore them until my task is completed,"No, i do not use them"
2,2,45 and above,Female,Graduate,Professional,4–6,Smartphone,Academic/Work-related,"Social Media (e.g., Facebook, Instagram, Linke...",Afternoon (12 PM–6 PM),I can work in any environment,Moderately productive,10–30 minutes,Take regular breaks,Check them briefly and resume my work,"No, i do not use them"
3,3,25–34,Male,Undergraduate,Professional,8-10,Laptop/PC,Academic/Work-related,"Social Media (e.g., Facebook, Instagram, Linke...",Afternoon (12 PM–6 PM),Quite workplace,"Extremely productive, i efficiently complete m...",10–30 minutes,"None, i prefer to work without any strategies",Ignore them until my task is completed,"No, i do not use them"
4,4,45 and above,Male,Graduate,Professional,8-10,Laptop/PC,Academic/Work-related,"Productivity (e.g., Microsoft Office, Notion)",Afternoon (12 PM–6 PM),I can work in any environment,"Extremely productive, i efficiently complete m...",30–60 minutes,"None, i prefer to work without any strategies",Check them briefly and resume my work,"No, i do not use them"


## 1. Preparación de los datos

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 16 columns):
 #   Column                      Non-Null Count  Dtype 
---  ------                      --------------  ----- 
 0   Unnamed: 0                  200 non-null    int64 
 1   Age Group                   200 non-null    object
 2   Gender                      200 non-null    object
 3   Education Level             200 non-null    object
 4   Occupation                  200 non-null    object
 5   Average Screen Time         200 non-null    object
 6   Device                      200 non-null    object
 7   Screen Activity             200 non-null    object
 8   App Category                200 non-null    object
 9   Screen Time Period          200 non-null    object
 10  Environment                 198 non-null    object
 11  Productivity                200 non-null    object
 12  Attention Span              200 non-null    object
 13  Work Strategy               197 non-null    object

In [ ]:
#Se realiza un conteo de valores nulos por variable en porcentajes
(df.isnull().mean() * 100).round(2)

,0
Unnamed: 0,0.0
Age Group,0.0
Gender,0.0
Education Level,0.0
Occupation,0.0
Average Screen Time,0.0
Device,0.0
Screen Activity,0.0
App Category,0.0
Screen Time Period,0.0


In [ ]:
# Se observa que hay una columna innecesaria creada llamada unnamed 0, la cual se procede a eliminarse.
df = df.drop(columns=["Unnamed: 0"])

# Tambien se observan algunos valores nulos, pero al ser pocos los registros (200 personas), esto equivale a: Environment = 1% (2 personas), Work Strategy = 1.5% (3 personas), Notification Handling = 0.5% (1 persona)
# Se proceden a eliminar los registros que contienen elementos vacios
df = df.dropna()

# Confirmar
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 196 entries, 0 to 199
Data columns (total 15 columns):
 #   Column                      Non-Null Count  Dtype 
---  ------                      --------------  ----- 
 0   Age Group                   196 non-null    object
 1   Gender                      196 non-null    object
 2   Education Level             196 non-null    object
 3   Occupation                  196 non-null    object
 4   Average Screen Time         196 non-null    object
 5   Device                      196 non-null    object
 6   Screen Activity             196 non-null    object
 7   App Category                196 non-null    object
 8   Screen Time Period          196 non-null    object
 9   Environment                 196 non-null    object
 10  Productivity                196 non-null    object
 11  Attention Span              196 non-null    object
 12  Work Strategy               196 non-null    object
 13  Notification Handling       196 non-null    object
 14 

In [ ]:
#Se realiza un conteo de valores nulos por variable en porcentajes
(df.isnull().mean() * 100).round(2)

,0
Age Group,0.0
Gender,0.0
Education Level,0.0
Occupation,0.0
Average Screen Time,0.0
Device,0.0
Screen Activity,0.0
App Category,0.0
Screen Time Period,0.0
Environment,0.0


In [ ]:
# Se ven los tipos de valores unicos que contiene cada columna
for col in df.columns:
    print(f"\nColumna: {col}")
    print(df[col].unique())


Columna: Age Group
['18–24' '45 and above' '25–34' 'Below 18' '35–44']

Columna: Gender
['Male' 'Female']

Columna: Education Level
['Undergraduate' 'Graduate' 'High school or below']

Columna: Occupation
['Student' 'Professional']

Columna: Average Screen Time
['More than 10' '8-10' '4–6' '6–8' '2–4' 'Less than 2']

Columna: Device
['Smartphone' 'Laptop/PC' 'Television' 'Tablet']

Columna: Screen Activity
['Entertainment (gaming, streaming, social media, etc.)'
 'Academic/Work-related']

Columna: App Category
['Social Media (e.g., Facebook, Instagram, LinkedIn, Twitter)'
 'Streaming (e.g., YouTube, Netflix)'
 'Productivity (e.g., Microsoft Office, Notion)' 'Gaming'
 'Messaging (e.g., WhatsApp, Messenger)']

Columna: Screen Time Period
['Evening (6 PM–10 PM)' 'Late night (10 PM–6 AM)' 'Afternoon (12 PM–6 PM)'
 'Morning (6 AM–12 PM)']

Columna: Environment
['Quite workplace' 'I can work in any environment'
 'Collaborative/team setting' 'Background noise/music']

Columna: Productivity
[

## 2.Transformación de Variables

In [ ]:
# Average Screen Time: Posee valores del tipo ['More than 10' '8-10' '4–6' '6–8' '2–4' 'Less than 2']. Se convierte a variable del tipo numerica basandose en el promedio de ambos valores de cada rango.
screen_map = {
    "Less than 2": 1,
    "2–4": 3,
    "4–6": 5,
    "6–8": 7,
    "8-10": 9,
    "More than 10": 11
}
df["Screen_Time_Hours"] = df["Average Screen Time"].map(screen_map)

In [ ]:
# Attention Span: Se convierte a una variable ordinal numerica ya que posee valores del tipo ['10–30 minutes' 'More than 1 hour' '30–60 minutes' 'Less than 10 minutes']

attention_map = {
    "Less than 10 minutes": 1,
    "10–30 minutes": 2,
    "30–60 minutes": 3,
    "More than 1 hour": 4
}
df["Attention_Score"] = df["Attention Span"].map(attention_map)

In [ ]:
# Productivity: Se convierte a variable ordinal.
productivity_map = {
    "Unproductive, i might not have completed the task and got carried away": 1,
    "Moderately productive": 2,
    "Extremely productive, i efficiently complete my tasks": 3
}

df["Productivity_Score"] = df["Productivity"].map(productivity_map)

In [ ]:
# Age Group: Se convierte a una categoria ordenada
age_order = ["Below 18", "18–24", "25–34", "35–44", "45 and above"]

df["Age Group"] = pd.Categorical(
    df["Age Group"],
    categories=age_order,
    ordered=True
)

In [ ]:
# Gemder: Se convierte a categoría simple
df["Gender"] = df["Gender"].astype("category")

In [ ]:
# Education Level: Se convierte a variable ordinal
edu_order = ["High school or below", "Undergraduate", "Graduate"]

df["Education Level"] = pd.Categorical(
    df["Education Level"],
    categories=edu_order,
    ordered=True
)

In [ ]:
# Ocupation: A categoría
df["Occupation"] = df["Occupation"].astype("category")

In [ ]:
# Device, Screen Activity, App Category, Screen Time Period: Todas a catergoricas nominales
cols_nominal = [
    "Device",
    "Screen Activity",
    "App Category",
    "Screen Time Period",
    "Environment",
    "Work Strategy",
    "Notification Handling"
]

for col in cols_nominal:
    df[col] = df[col].astype("category")

In [ ]:
# Usage of Productivity Apps: Se transfroma a Ordinales
apps_map = {
    "No, i do not use them": 0,
    "Yes, but i did not find them of any help": 1,
    "Yes, they are extremely helpful": 2
}

df["Productivity_App_Score"] = df["Usage of Productivity Apps"].map(apps_map)

In [ ]:
# Se revisan nuevamente los valores unicos de cada variable
for col in df.columns:
    print(f"\nColumna: {col}")
    print(df[col].unique())


Columna: Age Group
['18–24', '45 and above', '25–34', 'Below 18', '35–44']
Categories (5, object): ['Below 18' < '18–24' < '25–34' < '35–44' < '45 and above']

Columna: Gender
['Male', 'Female']
Categories (2, object): ['Female', 'Male']

Columna: Education Level
['Undergraduate', 'Graduate', 'High school or below']
Categories (3, object): ['High school or below' < 'Undergraduate' < 'Graduate']

Columna: Occupation
['Student', 'Professional']
Categories (2, object): ['Professional', 'Student']

Columna: Average Screen Time
['More than 10' '8-10' '4–6' '6–8' '2–4' 'Less than 2']

Columna: Device
['Smartphone', 'Laptop/PC', 'Television', 'Tablet']
Categories (4, object): ['Laptop/PC', 'Smartphone', 'Tablet', 'Television']

Columna: Screen Activity
['Entertainment (gaming, streaming, social medi..., 'Academic/Work-related']
Categories (2, object): ['Academic/Work-related', 'Entertainment (gaming, streaming, social medi...]

Columna: App Category
['Social Media (e.g., Facebook, Instagram,

Las variables claves con las que se van a trabajar son:

*   Screen_Time_Hours (promedio diario de tiempo frente a pantallas): 11 , 9 , 5 , 7 , 3 or 1 hours.
*   Attention_Score (duración de la atención autopercibida durante tareas digitales):
    *   "Less than 10 minutes": 1
    *   "10–30 minutes": 2
    *   "30–60 minutes": 3
    *   "More than 1 hour": 4
*   Productivity_Score (nivel de productividad autopercibido al realizar tareas basadas en pantalla):
    *   "Unproductive, i might not have completed the task and got carried away": 1
    *   "Moderately productive": 2,
    *   "Extremely productive, i efficiently complete my tasks": 3
*   Age Group (grupo etario del participante):
    *   Below 18 years old
    *   18–24 years old
    *   25–34 years old
    *   35–44 years old
    *   45 and above years old
*   Screen Time Period (momento del día en que se concentra el uso de pantalla):
    *   Afternoon (12 PM–6 PM)
    *   Evening (6 PM–10 PM)
    *   Late night (10 PM–6 AM)
    *   Morning (6 AM–12 PM)
*   Notification Handling (forma en que el participante gestiona notificaciones):
    *   Check them briefly and resume my work
    *   Ignore them until my task is completed
    *   Spend time interacting with the notifications
    *   Turn off notifications altogether

## 3. Gráficos y Correlaciones

### 3.1. Tiempo en pantalla vs capacidad de atención

In [ ]:
df[["Screen_Time_Hours", "Attention_Score"]].corr()

,Screen_Time_Hours,Attention_Score
Screen_Time_Hours,1.000000,0.096242
Attention_Score,0.096242,1.000000


Correlación entre el Tiempo en pantalla y capacidad de atención: 0.096.

Conclusión: La correlacion es muy débil, positiva y prácticamente inexistente, por lo tanto *no hay evidencia lineal clara entre el tiempo en pantalla con la capacidad de atención*

### 3.2. Tiempo en pantalla vs Productividad

In [ ]:
df[["Screen_Time_Hours", "Productivity_Score"]].corr()

,Screen_Time_Hours,Productivity_Score
Screen_Time_Hours,1.000000,0.010637
Productivity_Score,0.010637,1.000000


Correlación entre el Tiempo en Pantalla y la Productividad de la persona: 0.010.

Conclusión: *Tampoco hay una correlación entre el tiempo en pantalla con la productividad*

### 3.3. Promedio de tiempo en pantalla por grupo etáreo

In [ ]:
df.groupby("Age Group")["Screen_Time_Hours"].mean()

/tmp/ipython-input-960375800.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  df.groupby("Age Group")["Screen_Time_Hours"].mean()


,Screen_Time_Hours
Age Group,
Below 18,5.941176
18–24,7.098361
25–34,8.000000
35–44,7.200000
45 and above,7.666667


Promedio de tiempo de pantalla por grupo etáreo:
*   Menos de 18 años : 5.94 horas
*   Entre 18–24 años: 7.09 horas
*   Entre 25–34 años: 8.00 horas
*   Entre 35–44 años: 7.20 horas
*   Más de 45 años: 7.66 horas

Conclusiones:
*   *Los menores no son los que más usan pantalla.*
*   *El grupo entre 25–34 años tiene el promedio más alto.*





### 3.4. Momento de uso de pantalla en el dia vs capacidad de atención

In [ ]:
df.groupby("Screen Time Period")["Attention_Score"].mean()

/tmp/ipython-input-3447513504.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  df.groupby("Screen Time Period")["Attention_Score"].mean()


,Attention_Score
Screen Time Period,
Afternoon (12 PM–6 PM),2.640000
Evening (6 PM–10 PM),2.425532
Late night (10 PM–6 AM),2.800000
Morning (6 AM–12 PM),2.333333


Promedio de Capacidad de atención de acuerdo al momento del dia en el que se usa la pantalla:
*   Tarde (Afternoon): 2.64
*   Noche (Evening)
*   Trasnoche (Late Night): 2.8
*   Mañana (Morning): 2.33

Recordando que:
1 = Muy baja, 4 = Alta.

Conlusiones:
*  *El uso de pantalla por la noche NO muestra menor atención*.
*  *El menor promedio de capacidad de atención está en la mañana* (2.33)
*  *La percepción de atención no está directamente determinada por el tiempo o el horario de uso.*

### 3.5. Productividad vs Manejo de notificaciones

In [ ]:
df.groupby("Notification Handling")["Productivity_Score"].mean()

/tmp/ipython-input-2539033859.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  df.groupby("Notification Handling")["Productivity_Score"].mean()


,Productivity_Score
Notification Handling,
Check them briefly and resume my work,2.000000
Ignore them until my task is completed,2.418605
Spend time interacting with the notifications,1.444444
Turn off notifications altogether,2.125000


La productividad de acuerdo a la forma en la que la persona maneja las notificaciones:
*   Check briefly and resume: 2.00
*   Ignore until task completed: 2.42
*   Spend time interacting: 1.44
*   Turn off notifications: 2.12

Recordando que Productivity_Score:
1 = Unproductive,
2 = Moderate,
3 = Extremely productive.

Conlclusiones:
*   La productividad más baja (1.44) está en quienes: “Interactuan con las notificaciones”
*   La productividad más alta (2.42) está en quienes: “Ignoran las notificaciones hasta que terminan la tarea que estan realizando”
*   *El problema no es el tiempo total de uso de pantalla sino en la interrupción activa.*
*   L*as personas que: interactúan con notificaciones y cambian constantemente de foco, tienen peor rendimiento percibido.*




### 3.6. Estrategia de trabajo vs Capacidad de atención

In [ ]:
df.groupby("Work Strategy")["Attention_Score"].mean()

/tmp/ipython-input-1228206035.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  df.groupby("Work Strategy")["Attention_Score"].mean()


,Attention_Score
Work Strategy,
Eliminate all distractions,2.413043
"None, i prefer to work without any strategies",2.788732
Take regular breaks,2.358209
"Use some productivity tool (e.g, timers, blockers)",2.750000


Capacidad de atención agrupada por la estrategia de trabajo que opta la persona:
*   Eliminate distractions: 2.41
*   None: 2.79
*   Take breaks: 2.36
*   Use productivity tool: 2.75

Recordando que Attention_Score:
1 = Muy baja,
4 = Muy alta.

Conclusiones:
*   *No hay evidencia clara de que usar herramientas mejore dramáticamente la atención.*
*   Las diferencias son pequeñas para obtener mayores conclusiones.

### 3.7. Barplot: Cantidad de personas por tiempo de uso de pantalla por dia.

In [ ]:
import plotly.express as px
import pandas as pd

# Orden explícito para categorías (para que el eje X salga “bien”)
age_order = ["Below 18", "18–24", "25–34", "35–44", "45 and above"]
screen_order = ["Less than 2", "2–4", "4–6", "6–8", "8-10", "More than 10"]
period_order = ["Morning (6 AM–12 PM)", "Afternoon (12 PM–6 PM)", "Evening (6 PM–10 PM)", "Late night (10 PM–6 AM)"]


In [ ]:
counts = (
    df["Average Screen Time"]
    .value_counts(normalize=True)
    .reindex(screen_order)
    .reset_index()
)

counts.columns = ["Average Screen Time", "Percentage"]
counts["Percentage"] = counts["Percentage"] * 100

fig1 = px.bar(
    counts,
    x="Average Screen Time",
    y="Percentage",
    text=counts["Percentage"].round(1).astype(str) + "%",
    title="Distribución de tiempo de pantalla diario (%)",
    color_discrete_sequence=["#1f77b4"]
)

fig1.update_traces(
    textposition="outside",
    textfont=dict(size=20)
)

fig1.update_layout(
    template="simple_white",
    title=dict(
        text="Distribución de tiempo de pantalla diario (%)",
        x=0.5,  # 🔹 centrar título
        font=dict(size=24)
    ),
    xaxis=dict(
        title="Horas de pantalla/día",
        title_font=dict(size=24),
        tickfont=dict(size=20)
    ),
    yaxis=dict(
        title="Porcentaje de personas",
        title_font=dict(size=24),
        tickfont=dict(size=20),
        range=[0, counts["Percentage"].max() + 5]
    )
)

fig1.show()



Conclusión:
*   La mayoría de los participantes reportan entre 4 y 8 horas diarias de uso de pantalla, lo que refleja una exposición digital elevada y normalizada en la vida cotidiana.

### 3.8. Boxplot: Tiempo de pantalla vs Capacidad de Atención

In [ ]:
fig2 = px.box(
    df,
    x="Average Screen Time",
    y="Attention_Score",
    category_orders={"Average Screen Time": screen_order},
    points="all",
    title="Capacidad de Atención autopercibida según tiempo de pantalla por día",
    color_discrete_sequence=["#1f77b4"]
)

fig2.update_layout(
    template="simple_white",  # 🔹 fondo blanco
    title=dict(
        text="Atención vs Tiempo de Pantalla",
        x=0.5,  # 🔹 centrar título
        font=dict(size=24)
    ),
    xaxis=dict(
        title="Tiempo de pantalla (hs/dia)",
        title_font=dict(size=24),
        tickfont=dict(size=20)
    ),
    yaxis=dict(
        title="Capacidad de Atención (1=bajo, 4=alto)",
        title_font=dict(size=24),
        tickfont=dict(size=20)
    )
)

fig2.update_traces(
    marker=dict(size=6)
)

fig2.show()



Conclusiónes:
*   Las medianas de atención son bastante similares entre rangos.
*   No se observa una caída progresiva de la atención al aumentar las horas.
*   *Los rangos 8–10 y Más de 10 horas, no muestran menor atención que los intermedios.*
*  *No se observa una relación clara entre mayor tiempo de pantalla y menor atención autopercibida.*

### 3.9. Boxplot: Tiempo de pantalla vs Productividad

In [ ]:
fig3 = px.box(
    df,
    x="Average Screen Time",
    y="Productivity_Score",
    category_orders={"Average Screen Time": screen_order},
    points="all",
    title="Productividad vs Tiempo de Pantalla",
    color_discrete_sequence=["#1f77b4"]
)

fig3.update_layout(
    template="simple_white",  # 🔹 fondo blanco
    title=dict(
        text="Productividad vs Tiempo de Pantalla",
        x=0.5,  # 🔹 centrar título
        font=dict(size=24)
    ),
    xaxis=dict(
        title="Tiempo de pantalla (hs/dia)",
        title_font=dict(size=24),
        tickfont=dict(size=20)
    ),
    yaxis=dict(
        title="Productividad (1=bajo, 3=alto)",
        title_font=dict(size=24),
        tickfont=dict(size=20)
    )
)

fig3.update_traces(
    marker=dict(size=6)
)

fig3.show()


A medida que aumentan las horas de pantalla:


*   No se observa una caída progresiva en la mediana de productividad.
*   Los grupos de mayor uso (8–10 y More than 10) no muestran menor productividad que los intermedios.

Conclusiones:
*   *Mayor tiempo de pantalla = no menor productividad autopercibida.*


### 3.10. Barplot: Manejo de Notificaciones vs Productividad

In [ ]:
notif_mean = (
    df.groupby("Notification Handling", as_index=False)["Productivity_Score"]
    .mean()
    .sort_values("Productivity_Score")
)

fig4 = px.bar(
    notif_mean,
    x="Notification Handling",
    y="Productivity_Score",
    text=notif_mean["Productivity_Score"].round(2),
    title="La gestión de notificaciones se asocia con cambios en la productividad autopercibida",
    color_discrete_sequence=["#1f77b4"]
)

fig4.update_traces(textposition="outside")

fig4.update_layout(
    xaxis_title="Estrategia ante notificaciones",
    yaxis_title="Productividad (1=bajo, 3=alto)"
)

fig4.update_xaxes(
    ticktext=[
        "Interactúa con notificaciones",
        "Revisa brevemente y continúa",
        "Desactiva notificaciones",
        "Ignora hasta terminar"
    ],
    tickvals=[
        "Spend time interacting with the notifications",
        "Check them briefly and resume my work",
        "Turn off notifications altogether",
        "Ignore them until my task is completed"
    ],
    tickangle=0
)

fig4.show()

/tmp/ipython-input-2753195121.py:2: FutureWarning:

The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.



Conclusiones:
*   La forma en que se manejan las notificaciones se asocia con diferencias claras en productividad autopercibida.
*   El grupo que “interactúa con notificaciones” muestra el valor más bajo con diferencia.
*   Posponer o bloquear interrupciones = mayor productividad

*La fragmentación del foco parece tener mayor asociación con productividad que la cantidad de horas de exposición.*





### 3.11. Barplot: Estrategia de trabajo vs Capacidad de Atención

In [ ]:
work_mean = (
    df.groupby("Work Strategy", as_index=False)["Attention_Score"]
    .mean()
    .sort_values("Attention_Score")
)

fig5 = px.bar(
    work_mean,
    x="Work Strategy",
    y="Attention_Score",
    text=work_mean["Attention_Score"].round(2),
    title="Capacidad de Atención según Estrategia de Trabajo",
    color_discrete_sequence=["#1f77b4"]
)

fig5.update_traces(textposition="outside")

fig5.update_layout(
    xaxis_title="Estrategia de trabajo",
    yaxis_title="Capacidad de atención (1=bajo, 4=alto)"
)

# 🔹 Reemplazar etiquetas en español
fig5.update_xaxes(
    ticktext=[
        "Sin estrategia",
        "Toma descansos regulares",
        "Elimina distracciones",
        "Usa herramientas de productividad"
    ],
    tickvals=[
        "None, i prefer to work without any strategies",
        "Take regular breaks",
        "Eliminate all distractions",
        "Use some productivity tool (e.g, timers, blockers)"
    ],
    tickangle=0
)

fig5.show()

/tmp/ipython-input-2669608126.py:2: FutureWarning:

The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.



Conclusiones:
*   La estrategia de trabajo declarada no parece generar cambios drásticos en la atención autopercibida.
*   *El grupo “Sin estrategia” muestra la media más alta *(2.79). Esto es interesante porque contradice la idea de que usar estrategias formales necesariamente mejora la atención.
*   *Herramientas digitales muestran buen desempeño (pero no el mejor)*
*   Descansos y eliminación de distracciones no muestran ventaja clara




### 3.12. Barplot: Rango etareo vs Tiempo de pantalla

In [ ]:
age_mean = (
    df.groupby("Age Group", as_index=False)["Screen_Time_Hours"]
    .mean()
)

fig6 = px.bar(
    age_mean,
    x="Age Group",
    y="Screen_Time_Hours",
    category_orders={"Age Group": age_order},
    text=age_mean["Screen_Time_Hours"].round(2),
    title="Promedio de horas diarias de pantalla según grupo etario",
    color_discrete_sequence=["#1f77b4"]
)

fig6.update_traces(
    textposition="outside",
    textfont=dict(size=20)
)

fig6.update_layout(
    template="simple_white",  # 🔹 fondo blanco
    title=dict(
        text="Promedio de horas diarias de pantalla según grupo etario",
        x=0.5,  # 🔹 centrar título
        font=dict(size=24)
    ),
    xaxis=dict(
        title="Grupo etario",
        title_font=dict(size=24),
        tickfont=dict(size=20)
    ),
    yaxis=dict(
        title="Horas promedio de pantalla/día",
        title_font=dict(size=24),
        tickfont=dict(size=20)
    )
)

fig6.show()


/tmp/ipython-input-680479912.py:2: FutureWarning:

The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.



Conclusiones:
*   En esta muestra, los menores no presentan el mayor nivel de exposición digital.
*   El mayor consumo está en adultos jóvenes (25–34). La mayor exposición parece asociarse más a etapas de vida laboral activa.
*   No hay un salto dramático entre generaciones.

### 3.13. Barplot: Manejo de Notificaciones vs Capacidad de Atención

In [ ]:
notif_att = (
    df.groupby("Notification Handling", as_index=False)["Attention_Score"]
    .mean()
    .sort_values("Attention_Score")
)

fig7 = px.bar(
    notif_att,
    x="Notification Handling",
    y="Attention_Score",
    text=notif_att["Attention_Score"].round(2),
    title="Manejo de Notificaciones y Capacidad de Atención",
    color_discrete_sequence=["#1f77b4"]
)

fig7.update_traces(textposition="outside")

fig7.update_layout(
    xaxis_title="Estrategia ante notificaciones",
    yaxis_title="Capacidad de atención promedio(1=bajo, 4=alto)"
)

fig7.update_xaxes(
    ticktext=[
        "Interactúa con notificaciones",
        "Revisa brevemente y continúa",
        "Desactiva notificaciones",
        "Ignora hasta terminar"
    ],
    tickvals=[
        "Spend time interacting with the notifications",
        "Check them briefly and resume my work",
        "Turn off notifications altogether",
        "Ignore them until my task is completed"
    ],
    tickangle=0
)

fig7.show()

/tmp/ipython-input-3639200534.py:2: FutureWarning:

The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.



Conclusiones:
*   A medida que la estrategia protege más el foco de la atención, la atención
promedio aumenta.
*   El grupo que se detiene a interactuar muestra el nivel más bajo de atención. Ignorar las notificaciones hasta terminar la tarea es la estrategia más asociada a mayor atención
*   El problema no parece ser la cantidad de tiempo frente a pantalla, sino la fragmentación atencional generada por interrupciones digitales frecuentes.

In [ ]:
notif_order = (
    df.groupby("Notification Handling")["Attention_Score"]
    .mean()
    .sort_values()
    .index
    .tolist()
)

fig = px.box(
    df,
    x="Notification Handling",
    y="Attention_Score",
    category_orders={"Notification Handling": notif_order},
    points="all",
    title="Atención vs Manejo de notificaciones",
    color_discrete_sequence=["#1f77b4"]
)

fig.update_layout(
    template="simple_white",
    title=dict(
        text="Atención vs Manejo de Notificaciones",
        x=0.5,
        font=dict(size=24)
    ),
    xaxis=dict(
        title="Manejo de notificaciones",
        title_font=dict(size=24),
        tickfont=dict(size=20)
    ),
    yaxis=dict(
        title="Capacidad de atención (1=bajo, 4=alto)",
        title_font=dict(size=24),
        tickfont=dict(size=20),
        range=[0.8, 4.2],
        dtick=1
    )
)

fig.update_xaxes(
    ticktext=[
        "Interactúa con notificaciones",
        "Revisa brevemente y continúa",
        "Desactiva notificaciones",
        "Ignora hasta terminar"
    ],
    tickvals=[
        "Spend time interacting with the notifications",
        "Check them briefly and resume my work",
        "Turn off notifications altogether",
        "Ignore them until my task is completed"
    ],
    tickangle=0
)

fig.update_traces(marker=dict(size=6))

fig.show()


/tmp/ipython-input-119330904.py:2: FutureWarning:

The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.



### 3.13. Heatmap: Momento del dia de uso de pantalla vs Manejo de notificaciones (Productividad como variable de color)

In [ ]:
fig = px.imshow(
    pivot,
    text_auto=True,
    aspect="auto",
    color_continuous_scale="RdYlGn",
    title="Productividad promedio según horario del día y manejo de notificaciones"
)

fig.update_layout(
    template="simple_white",
    title_font=dict(size=22),
    xaxis_title="Estrategia ante notificaciones",
    yaxis_title="Horario del día"
)


fig.update_xaxes(
    title="Estrategia ante notificaciones",
    title_font=dict(size=24),
    tickfont=dict(size=20),
    tickangle=0
)

fig.update_yaxes(
    title="Horario del día",
    title_font=dict(size=24),
    tickfont=dict(size=20)
)


fig.update_layout(
    coloraxis_colorbar=dict(
        title="Productividad",
        tickfont=dict(size=20),
        title_font=dict(size=20)
    )
)

fig.show()


NameError: name 'pivot' is not defined

Eje Y: Horario de uso
Eje X: Manejo de notificaciones
Color: Productividad promedio (1= menos, 3= Mas)

Conclusiones:
*   La columna "Interactúa con Notificaciones" tiene consistentemente los valores más bajos.
*   Mientras que "Ignora hasta terminar" muestra valores más altos de productividad.

Independientemente del momento del día, *los participantes que interactúan activamente con notificaciones reportan menores niveles de productividad*. En contraste, *quienes posponen o ignoran notificaciones muestran consistentemente mayores niveles de rendimiento autopercibido.*

### 3.14. Box plot: Productividad vs. Manejo de Notificaciones

In [ ]:
notif_order = (
    df.groupby("Notification Handling")["Productivity_Score"]
    .mean()
    .sort_values()
    .index
    .tolist()
)

fig = px.box(
    df,
    x="Notification Handling",
    y="Productivity_Score",
    category_orders={"Notification Handling": notif_order},
    points="all",
    title="Productividad vs Manejo de Notificaciones",
    color_discrete_sequence=["#1f77b4"]
)


fig.update_layout(
    template="simple_white",
    title=dict(
        text="Productividad vs Manejo de Notificaciones",
        x=0.5,
        font=dict(size=24)
    ),
    xaxis=dict(
        title="Manejo de Notificaciones",
        title_font=dict(size=24),
        tickfont=dict(size=20)
    ),
    yaxis=dict(
        title="Productividad (1=bajo, 3=alto)",
        title_font=dict(size=24),
        tickfont=dict(size=20)
    )
)


fig.update_xaxes(
    ticktext=[
        "Interactúa con notificaciones",
        "Revisa brevemente y continúa",
        "Desactiva notificaciones",
        "Ignora hasta terminar"
    ],
    tickvals=[
        "Spend time interacting with the notifications",
        "Check them briefly and resume my work",
        "Turn off notifications altogether",
        "Ignore them until my task is completed"
    ],
    tickangle=0
)


fig.update_traces(marker=dict(size=6))

fig.show()


/tmp/ipython-input-1922255313.py:2: FutureWarning:

The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.



Conclusiones:
*   “Interactúa con notificaciones” muestra la peor distribución: Interactuar activamente con notificaciones se asocia con menor productividad autopercibida de forma consistente, no solo en promedio sino en distribución.
*   “Ignora hasta terminar” muestra la mejor consistencia: Proteger el foco hasta completar la tarea se asocia con mayor productividad de manera consistente en la muestra.
*   “Desactiva notificaciones” también muestra buen rendimiento: Esto refuerza que el control de interrupciones tiene efecto observable.
*   Revisa brevemente y continúa” queda en el medio: Las micro-interrupciones parecen tener un costo intermedio.





### 3.15. Entretenimiento vs uso académico

In [ ]:
df.groupby("Screen Activity")["Productivity_Score"].mean()
df.groupby("Screen Activity")["Attention_Score"].mean()


/tmp/ipython-input-3005752555.py:1: FutureWarning:

The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.

/tmp/ipython-input-3005752555.py:2: FutureWarning:

The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.



,Attention_Score
Screen Activity,
Academic/Work-related,2.651163
"Entertainment (gaming, streaming, social media, etc.)",2.472727


Capacidad de atención respecto al nivel académico:
*   Academic/Work-related: 2.65
*   Entertainment: 2.47

Conclusión:
*   Uso orientado a tarea, tiene mayor atención autopercibida
*   Uso orientado a entretenimiento, tiene menor atención autopercibida

### 3.16. Categoría de app vs atención

In [ ]:
df.groupby("App Category")["Attention_Score"].mean().sort_values()


/tmp/ipython-input-173023829.py:1: FutureWarning:

The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.



,Attention_Score
App Category,
"Social Media (e.g., Facebook, Instagram, LinkedIn, Twitter)",2.371134
"Messaging (e.g., WhatsApp, Messenger)",2.400000
"Streaming (e.g., YouTube, Netflix)",2.521739
"Productivity (e.g., Microsoft Office, Notion)",2.895833
Gaming,3.125000


Capacidad de atención de acuerdo a la aplicacion que se usa:
*   Social Media: 2.37
*   Messaging: 2.4
*   Streaming: 2.52
*   Productivity: 2.90
*   Gaming: 3.12

Conslusiones:
*   Las apps asociadas a interacción social rápida y notificaciones constantes dan una menor atención autopercibida. Por lo tanto no todo el uso recreativo fragmenta la atención de la misma manera.
      *   El gaming requiere foco sostenido.
      *   Las redes sociales requieren atención fragmentada.

*   Apps de productividad e incluso gaming dan una mayor atención autopercibida.

In [ ]:

app_labels = {
    "Social Media (e.g., Facebook, Instagram, LinkedIn, Twitter)": "Redes sociales",
    "Messaging (e.g., WhatsApp, Messenger)": "Mensajería",
    "Streaming (e.g., YouTube, Netflix)": "Streaming",
    "Productivity (e.g., Microsoft Office, Notion)": "Productividad",
    "Gaming": "Gaming"
}

fig.update_xaxes(
    ticktext=[
        "Redes sociales",
        "Mensajería",
        "Streaming",
        "Productividad",
        "Gaming"
    ],
    tickvals=list(app_labels.keys()),
    tickangle=0,
    tickfont=dict(size=14)
)

fig.update_layout(
    template="simple_white",
    margin=dict(l=60, r=30, t=70, b=60)
)

fig.show()


### 3.18. Estudiantes vs profesionales

In [ ]:
df.groupby("Occupation")["Screen_Time_Hours"].mean()

/tmp/ipython-input-3481392627.py:1: FutureWarning:

The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.



,Screen_Time_Hours
Occupation,
Professional,8.000000
Student,6.442623


In [ ]:
df.groupby("Occupation")["Attention_Score"].mean()

/tmp/ipython-input-222965497.py:1: FutureWarning:

The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.



,Attention_Score
Occupation,
Professional,2.581081
Student,2.532787


In [ ]:
df.groupby("Occupation")["Productivity_Score"].mean()

/tmp/ipython-input-3387382736.py:1: FutureWarning:

The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.



,Productivity_Score
Occupation,
Professional,2.297297
Student,1.950820


Tempo en pantalla por ocupacion
*   Profesional: 8 horas
*   Estudiante: 6.44 horas

Atención promedio por ocupacion
*   Profesional: 2.58
*   Estudiante: 1.53

Productividad promedio agrupada por Ocupación:
*   Profesional: 2.3
*   Estudiante: 1.95

Conclusiones:
Los estudiantes:
    *   No usan más pantalla.
    *   No tienen menor atención autopercibida.
    *   Pero sí reportan menor productividad.

### 3.19. Productivity Apps — ¿sirven o no?

In [ ]:
df.groupby("Usage of Productivity Apps")["Productivity_Score"].mean()


,Productivity_Score
Usage of Productivity Apps,
"No, i do not use them",2.066667
"Yes, but i did not find them of any help",1.906250
"Yes, they are extremely helpful",2.250000


Productividad promediodependiendo de uso de apps de productividad:


*   No usan apps: 2.07
*   Las usan pero no ayudan: 1.91
*   Las usan y ayudan: 2.25

Conclusiones:


*   Quienes perciben que las apps son útiles reportan mayor productividad.
*   Quienes dicen que no ayudan reportan la menor productividad.Es decir que la productividad es más baja en quienes “Sí las usan, pero no ayudan.”
Eso sugiere:
      *   Frustración
      *   Uso ineficiente
      *   Dependencia sin estrategia
      *   O falta de integración real en el flujo de trabajo

Las herramientas digitales por sí solas no parecen garantizar mayor productividad. La diferencia emerge cuando el usuario percibe que la herramienta está alineada con su estrategia de trabajo.




### 3.20. Heavy Users vs Light Users

In [ ]:
df["Heavy_User"] = df["Screen_Time_Hours"] >= 8


In [ ]:
df.groupby("Heavy_User")["Attention_Score"].mean()

,Attention_Score
Heavy_User,
False,2.491935
True,2.652778


In [ ]:
df.groupby("Heavy_User")["Productivity_Score"].mean()

,Productivity_Score
Heavy_User,
False,2.072581
True,2.097222


Capacidad de atencion:
*   Light users = 2.49
*   Heavy users = 2.65

Productividad Promedio:
*   Light users = 2.07
*   Heavy users = 2.09

Conslusiones:


*   Más horas de pantalla = no menor atención (Incluso, en tu muestra, los heavy users muestran levemente mayor atención autopercibida.)
*   Más horas de pantalla = no menor productividad

In [ ]:
tmp = df.groupby("Heavy_User", as_index=False)[["Attention_Score","Productivity_Score"]].mean()
tmp_melt = tmp.melt(id_vars="Heavy_User", var_name="Metric", value_name="Score")
px.bar(tmp_melt, x="Metric", y="Score", color="Heavy_User", barmode="group")
